assets/wizards/newspack/views/settings/connections/google-oauth.tsx

```diff
@@ -0,0 +1,169 @@
+/**
+ * Settings Wizard: Connections > Google OAuth
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies.
+ */
+import { Button } from '../../../../../components/src';
+import WizardsActionCard from '../../../../wizards-action-card';
+import { useWizardApiFetch } from '../../../../hooks/use-wizard-api-fetch';
+import { WizardError, WIZARD_ERROR_MESSAGES } from '../../../../errors';
+
+function getURLParams() {
+	const searchParams = new URLSearchParams( window.location.search );
+	const params: { [ key: string ]: string } = {};
+	for ( const [ key, value ] of searchParams.entries() ) {
+		params[ key ] = value;
+	}
+	return params;
+}
+
+function GoogleOAuth( {
+	onSuccess,
+	isOnboarding,
+}: {
+	onInit?: ( str: Error | null ) => void;
+	onSuccess?: ( arg: OAuthData ) => void;
+	isOnboarding?: ( str: string ) => void;
+} ) {
+	const [ authState, setAuthState ] = useState< OAuthData >( {} );
+
+	const userBasicInfo = authState?.user_basic_info;
+	const isConnected = Boolean( userBasicInfo && userBasicInfo.email );
+
+	const {
+		setError,
+		errorMessage,
+		wizardApiFetch,
+		isFetching: inFlight,
+	} = useWizardApiFetch( '/newspack-settings/connections/apis/google-oauth' );
+
+	// We only want to autofetch the current auth state if we're not onboarding.
+	useEffect( () => {
+		if ( ! isOnboarding ) {
+			getCurrentAuth();
+		}
+	}, [] );
+
+	useEffect( () => {
+		if ( isConnected && userBasicInfo && ! userBasicInfo.has_refresh_token ) {
+			setError(
+				new WizardError(
+					WIZARD_ERROR_MESSAGES.GOOGLEOAUTH_REFRESH_TOKEN_EXPIRED,
+					'googleoauth_refresh_token_expired'
+				)
+			);
+		}
+	}, [ isConnected ] );
+
+	function getCurrentAuth() {
+		const params = getURLParams();
+		if ( ! params.access_token ) {
+			wizardApiFetch< OAuthData >(
+				{
+					path: '/newspack/v1/oauth/google',
+				},
+				{
+					onSuccess( data ) {
+						setAuthState( data );
+						if ( data?.user_basic_info ) {
+							if ( typeof onSuccess === 'function' ) {
+								onSuccess( data );
+							}
+						}
+					},
+				}
+			);
+		}
+	}
+
+	function openAuth() {
+		const authWindow = window.open(
+			'about:blank',
+			'newspack_google_oauth',
+			'width=500,height=600'
+		);
+		wizardApiFetch< string >(
+			{
+				path: '/newspack/v1/oauth/google/start',
+			},
+			{
+				onSuccess( url ) {
+					/** authWindow can be 'null' due to browser's popup blocker. */
+					if ( authWindow ) {
+						authWindow.location = url;
```

Comment:
On Firefox, the auth window opens but the location is never updated, and so I end up with a popup of the homepage:

![Screenshot 2024-06-24 at 16 04 39](https://github.com/Automattic/newspack-plugin/assets/17905991/1bc121a3-9b0c-4e55-8b80-c3b5500ec3ac)

We should probably not trigger the opening of the popup until the api fetch is successful and we are certain there are no popup blocker issues, etc.

Reply:
Thanks @jaredrethman! 

This is working as expected now. It looks like there was an issue with my local site config where the `NEWSPACK_GOOGLE_OAUTH_PROXY_OVERRIDE` was not correct which was causing the popup location not to update.

This did get me thinking though, right now if the popup fails to open and `authWindow` is null we simply abort with no additional error message. We should probably handle this case by notifying the publisher that they should disable any popup blocker, etc. and try again.

WDYT?

Reply:
@chickenn00dle great idea, I have gone ahead and added some error reporting for null `authWindow` and `url` values.